src/Microsoft.DotNet.XHarness.TestRunners.Xunit/WasmXmlResultWriter.cs

```diff
@@ -19,5 +19,7 @@ public static void WriteOnSingleLine(XElement assembliesElement)
             Console.WriteLine($"STARTRESULTXML {bytes.Count} {base64} ENDRESULTXML");
             Console.WriteLine($"Finished writing {bytes.Count} bytes of RESULTXML");
         }
+
+        public static string EscapeNewLines(string message) => message.Replace("\r", "\\r").Replace("\n", "\\n");

```

Comment:
we also have some data driven unit tests with nasty unicode codepoints. 
It makes test names, which the python downstream can't parse. 
Ending up with `�` when things are happy or losing the log if not.
I wonder if you want to also `String.Normalize` or something.

Reply:
does this happen with the standard xunit test runner too? I wonder if we can copy some normalization code from there.

Reply:
Looking at xunit source, I have found this smaller beast https://github.com/xunit/xunit/blob/2210888189e19227f4b0413f55146792f69bdf84/src/xunit.v3.runner.common/Reporters/DefaultRunnerReporterMessageHandler.cs#L98 and this bigger one https://github.com/xunit/xunit/blob/2210888189e19227f4b0413f55146792f69bdf84/src/xunit.v3.runner.common/Sinks/DelegatingSinks/DelegatingXmlCreationSink.cs#L400

Reply:
IIRC the line replacement was there to ensure that `WasmTestMessagesProcessor` (the server-side message pump) was seeing PASS/FAIL messages as a single line and thus as a single message and thus it was changing the scope (from info to debug) correctly. I have removed this scope changing logic in this PR, so I should be safe to remove the line replacement 🤞